modules/tide_share_link/src/Controller/ShareLinkTokenController.php

```diff
@@ -130,14 +130,11 @@ public function checkNodeAccess(AccountInterface $account) : AccessResultInterfa
 
     if ($node && $node instanceof NodeInterface) {
       // Attempt to retrieve the node revision.
-      $vid = $this->routeMatch->getParameter('node_revision');
-      if ($vid) {
-        $revision = $this->entityTypeManager()->getStorage('node')->loadRevision($vid);
-        if ($revision && $revision->id() === $node->id()) {
-          return AccessResult::allowedIf($revision->access('view', $account))
-            ->addCacheableDependency($revision)
-            ->addCacheableDependency($account);
-        }
+      $revision = $this->routeMatch->getParameter('node_revision');
+      if ($revision && $revision->id() === $node->id()) {
```

Comment:
this actually returns the revision id directly - `$revision = $this->routeMatch->getParameter('node_revision');`
Sp do not need to do this `$revision->id()` it ends up throwing error